File: Solution/CSharp/1. Two Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        if (nums == null || nums.Length < 1) {
            return null;
        }
        Dictionary<int, int> dict = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            if (dict.ContainsKey(target - nums[i])) {
                int[] arr = { dict[target - nums[i]], i };
                return arr;
            } else {
                dict[nums[i]] = i;
            }
        }
        return null;
    }
}

File: Solution/Python/1. Two Sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}
        for i in range(len(nums)):
            c = nums[i]
            if m.get(target - c) != None:
                return [m[target - c], i]
            else:
                m[c] = i
        return []


s = Solution()

#nums = [3, 2, 4]
#target = 6
nums = [3, 3]
target = 6
res = s.twoSum(nums, target)
print(res)
